### ISSUE TITLE
Sporadic shell crashes on GNOME 40

### ISSUE THREAD

**Issue by keeferrourke** [flathub/com.slack.Slack#117]
I'm not sure if this is the right place to report this, but I've been experiencing some GNOME-shell crashes (that is: the whole desktop crashes, not just Slack) that appear to be related to usage of this flatpak. I'm unsure *why* this happens, but some interactions in Slack will just crash the whole desktop.
I've observed this happen now under both X.org and Wayland display servers.


Details follow:

**Operating system**:
```
NAME=Fedora
VERSION="34 (Workstation Edition)"
ID=fedora
VERSION_ID=34
VERSION_CODENAME=""
PLATFORM_ID="platform:f34"
PRETTY_NAME="Fedora 34 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:34"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/34/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=34
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=34
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

**Flatpak version**:
Flatpak 1.10.2

**Slack version**: 
<details>
<summary>flatpak info com.slack.Slack (click to expand)</summary>

    $ flatpak info com.slack.Slack                                                                                                                                                                                                              

    Slack - Chat with your team

              ID: com.slack.Slack
             Ref: app/com.slack.Slack/x86_64/stable
            Arch: x86_64
          Branch: stable
         Version: 4.14.0
         License: LicenseRef-proprietary
          Origin: flathub
      Collection: org.flathub.Stable
    Installation: system
       Installed: 5.3 MB
         Runtime: org.freedesktop.Platform/x86_64/20.08
             Sdk: org.freedesktop.Sdk/x86_64/20.08

          Commit: f738078bef50d00d548abcaa88d6432dada3999582bc66b96e36154a6b62420a
          Parent: dfb8851acb9d73bf245ef0260d878fc09c65c9f96703f53ef4cf9d16e55d308c
         Subject: Update slack.deb to 4.14.0 (e0251447)
            Date: 2021-03-20 08:44:53 +0000

</details>

**Crash log from journalctl:**

See attached. This is the most recently observed crash under X.org, which happened shortly after launching Slack. [gnome40-shell-crash.txt](https://github.com/flathub/com.slack.Slack/files/6392080/gnome40-shell-crash.txt)

This line is highly suspect:

```
Apr 28 09:13:24 aipom gnome-shell[24124]: Object 0x55e99ae102a0 (a ClutterInputDevice) resurfaced after the JS wrapper was finalized. This is some library doing dubious memory management inside dispose()
```

If this is the wrong place to report this, I apologize. If I can provide any more useful information, please let me know.

**Comment by vchernin** [comment]
As a test, could you try running slack in a [chromium flatpak](https://github.com/flathub/com.github.Eloston.UngoogledChromium) and seeing if the behaviour is any different? This could help isolate whether its a slack flatpak issue or perhaps something more broad (like a general flatpak issue).

**Comment by keeferrourke** [comment]
I'm going to close this bug as it's likely an issue a couple steps upstream. 

 - A similar issue that was reported for Microsoft Teams [on the AppIndicator extension, here](https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/272).
 - Additionally, a similar report to this was filed against GJS https://gitlab.gnome.org/GNOME/gjs/-/issues/400

To see if it was the same issue, I disabled GNOME Shell extensions for a while and never experienced the crash again (at the expense of missing notifications from Slack). 

Work is being done to resolve this issue in GJS, and since I've updated the AppIndicator extension to version 37, I haven't experienced any more crashes from this flatpak.

Sorry for the noise!
